pkg/pod/pod.go

```diff
@@ -139,9 +139,12 @@ func (b *Builder) Build(ctx context.Context, taskRun *v1beta1.TaskRun, taskSpec
 		return nil, err
 	}
 
+	// Create Step timeout arguments
+	stepTimeoutArgs := returnTimeoutArgs(taskSpec.Steps)
+
 	// Rewrite steps with entrypoint binary. Append the entrypoint init
 	// container to place the entrypoint binary.
-	entrypointInit, stepContainers, err := orderContainers(b.Images.EntrypointImage, credEntrypointArgs, stepContainers, taskSpec.Results)
+	entrypointInit, stepContainers, err := orderContainers(b.Images.EntrypointImage, credEntrypointArgs, stepContainers, stepTimeoutArgs, taskSpec.Results)
```

Comment:
The args to `orderContainers` are becoming a bit of a smell. 👃 

Perhaps it should take a `[]Step` instead of `[]Container` and a parallel slice of timeouts?

(Not necessarily required for this change, but something to think about)

Reply:
Resolved this by adding the timeout to the stepContainers earlier on in the code.